### ISSUE TITLE
Runtime irrelevance

### ISSUE THREAD

**Issue by kurnevsky** [agda/agda-stdlib#2486]
Hi. I'm trying to use [erased annotations](https://agda.readthedocs.io/en/v2.7.0.1/language/runtime-irrelevance.html) but as I can see agda stdlib doesn't really support them. What I stumbled upon so far:
- `⊥-elim` can't take an erased `⊥` argument which is often needed when all proofs are erased.
- Idris defines `Σ` variants with erased first and second values called [Exists](https://github.com/idris-lang/Idris2/blob/0e83d6c7c6ad8b3b98758d6b5ab875121992c44c/libs/base/Data/DPair.idr#L63) and [Subset](https://github.com/idris-lang/Idris2/blob/0e83d6c7c6ad8b3b98758d6b5ab875121992c44c/libs/base/Data/DPair.idr#L102) which are also quite convenient to erase proofs.

Should maybe agda stdlib have something similar? That would require agda stdlib to have `--erasure` option enabled requires all dependent modules to have it.

**Comment by jamesmckinna** [comment]
See #2346 for an earlier version of a similar line of thought... I'd be broadly in favour, but for the complicated balance of forces/compatibility issues (esp. wrt infective options) involved in what constitutes a 'standard' library.

**Comment by gallais** [comment]
Note that you can implement
```
recover : @0 ⊥ → ⊥
recover ()
```
so you can plug your contradiction into a call to `⊥-elim`.

**Comment by jamesmckinna** [comment]
Cf. `Relation.Nullary.Recomputable.⊥-recompute`... ob @gallais 's observation

**Comment by kurnevsky** [comment]
> while your other examples from Idris seem to be related to the other combinators defined there...?

Is it a question to me? :)
I meant that it's sometimes useful to have dependent sum types with erased first/second values, and Idris has them. For instance, using my own definition of such type gave me almost 2x speed up in runtime for my code :)

**Comment by jamesmckinna** [comment]
See also: `Data.Refinement` for a version of "dependent sum types with erased second values" at least. 
Agree that erased *first* projections are independently interesting, though...

**Comment by gallais** [comment]
There's also the issue of `Irrelevant`  vs. `@0` annotation. I don't think we have anything `@0`-related in the lib at the minute.

**Comment by jamesmckinna** [comment]
> There's also the issue of `Irrelevant` vs. `@0` annotation. I don't think we have anything `@0`-related in the lib at the minute.

Absolutely! I raised this explicitly as a point for us to discuss after the merge of `experimental` following the v2.7.0 release...
